src/test/java/dev/snowdrop/release/services/CPaaSConfigUpdateServiceTest.java

```diff
@@ -0,0 +1,190 @@
+        try {
+            git.initRepository(config);
+            Stream<File> files = service.updateCPaaSFiles(RELEASE, git.getConfig(config).getRepository().getWorkTree(), PREVIOUS_RELEASE, false, true);
+            Map<String, File> fileMap = files.collect(Collectors.toMap(file -> file.getName(), file -> file));
+            assertTrue(fileMap.containsKey(PRODUCT_FILE_NAME), fileMap.toString());
+            testProductFile(factory.createCPaaSProductFromStream(new FileInputStream(fileMap.get(PRODUCT_FILE_NAME))));
+        } catch (IOException | ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            fail(e);
+        }
+    }
+
+    private void testProductFile(CPaaSProductFile productFile) {
+        CPaaSProduct product = productFile.getProduct();
+        assertEquals(RELEASE, product.getRelease().getVersion());
+        assertEquals("spring-boot/2.4", product.getProjects().get(0).getComponents().get(0).getBuilds().get(0).getPigSource().getRoot());
+    }
+
+    @Test
+    public void testReleaseCommon() {
+        final String token = ConfigProvider.getConfig().getValue("gitlab.token", String.class);
+        final String user = ConfigProvider.getConfig().getValue("gitlab.user", String.class);
+        GitService.GitConfig config = service.buildGitConfig(RELEASE, user, token, Optional.of(PREVIOUS_RELEASE), Optional.of(ConfigProvider.getConfig().getOptionalValue("gitlab.repository", String.class).orElse(user + "/springboot")));
+        try {
+            git.initRepository(config);
+            Stream<File> files = service.updateCPaaSFiles(RELEASE, git.getConfig(config).getRepository().getWorkTree(), PREVIOUS_RELEASE, false, true);
+            Map<String, File> fileMap = files.collect(Collectors.toMap(file -> file.getName(), file -> file));
+            assertTrue(fileMap.containsKey(RELEASE_FILE_NAME), fileMap.toString());
+            testReleaseFile(factory.createCPaaSReleaseFromStream(new FileInputStream(fileMap.get(RELEASE_FILE_NAME))));
+        } catch (IOException | ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            fail(e);
+        }
+    }
+
+    private void testReleaseFile(CPaaSReleaseFile releaseFile) {
+        CPaaSRelease release = releaseFile.getRelease();
+        List<CPaaSPipelines> pipelines = release.getPipelines();
+        assertTrue("build".equalsIgnoreCase(pipelines.get(0).getName()));
+        assertTrue("release".equalsIgnoreCase(pipelines.get(1).getName()));
+        CPaaSAdvisory cpaasAdvisoryRelease = release.getTools().get(0).getAdvisories().get(0);
+        assertTrue(cpaasAdvisoryRelease.getSynopsis().contains(RELEASE), releaseFile.toString());
+        assertFalse(cpaasAdvisoryRelease.getSynopsis().contains(PREVIOUS_RELEASE), releaseFile.toString());
+    }
+
+    @Test
+    public void testReleaseDr() {
```

Comment:
Same remarks as previously:
- name more descriptive
- add exeption to the method signature

Reply:
Done